lib/msf/core/exploit/dns/server.rb

```diff
@@ -0,0 +1,172 @@
+# -*- coding: binary -*-
+require 'msf/core'
+require 'rex/proto/dns'
+
+
+module Msf
+
+###
+#
+# This module exposes methods for querying a remote DNS service
+#
+###
+module Exploit::Remote::DNS
+module Server
+  include Exploit::Remote::SocketServer
+
+  MATCH_HOSTNAME = Rex::Proto::DNS::Constants::MATCH_HOSTNAME
+
+  #
+  # Initializes an exploit module that serves DNS requests
+  #
+  def initialize(info = {})
+    super
+
+    register_options(
+      [
+        OptPort.new('SRVPORT', [true, 'The local port to listen on.', 53]),
+        OptString.new('STATIC_ENTRIES', [ false, "DNS domain search list (hosts file or space/semicolon separate entries)"]),
+        OptBool.new('DISABLE_RESOLVER', [ false, "Disable DNS request forwarding", false]),
+        OptBool.new('DISABLE_NS_CACHE', [ false, "Disable DNS response caching", false])
+      ], Exploit::Remote::DNS::Server
+    )
+
+    register_advanced_options(
+      [
+        OptBool.new('DNS_SERVER_UDP', [true, "Serve UDP DNS requests", true]),
```

Comment:
So apologies for not enforcing this very well by us, so there's not enough consistency, but could you please make all your advanced options CamelCase?

It was documented here: https://github.com/rapid7/metasploit-framework/wiki/How-to-use-datastore-options#ideal-datastore-naming

Thanks.